src/main/java/com/boj/guidance/controller/MemberController.java

```diff
@@ -0,0 +1,82 @@
+package com.boj.guidance.controller;
+
+import com.boj.guidance.config.PasswordEncoder;
+import com.boj.guidance.dto.MemberJoinRequestDto;
+import com.boj.guidance.dto.MemberLoginRequestDto;
+import com.boj.guidance.dto.MemberResponseDto;
+import com.boj.guidance.service.MemberService;
+import com.boj.guidance.util.api.ApiResponse;
+import com.boj.guidance.util.api.ResponseCode;
+import jakarta.servlet.http.HttpSession;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@RestController
+@RequestMapping("/member")
+public class MemberController {
+
+    private final MemberService memberService;
+    private final PasswordEncoder passwordEncoder;
+
+    public MemberController(MemberService memberService, PasswordEncoder passwordEncoder) {
+        this.memberService = memberService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    /**
+     * 사용자 회원가입 기능
+     * @param dto
+     * @return
+     */
+    @RequestMapping(method = RequestMethod.POST, value = "/join")
+    public ApiResponse<MemberResponseDto> join(@RequestBody MemberJoinRequestDto dto) throws Exception {
+        log.info("회원가입 API 호출");
+        String encoded = passwordEncoder.encrypt(dto.getLogin_password());
+        MemberResponseDto joined = memberService.join(
+                MemberJoinRequestDto.builder()
+                        .handle(dto.getHandle())
+                        .login_id(dto.getLogin_id())
+                        .login_password(encoded)
+                        .bio(dto.getBio())
+                        .solved_count(dto.getSolved_count())
+                        .tier(dto.getTier())
+                        .rating(dto.getRating())
+                        .rating_by_problems_sum(dto.getRating_by_problems_sum())
+                        .rating_by_solved_count(dto.getRating_by_solved_count())
+                        .build()
+        );
```

Comment:
```suggestion
        Member member = memberService.join(dto.toEntity());
        MemberResponseDto joined = MemberResponseDto.of(member);
```

Request, Response에 쓰이는 Dto와 Entity로 쓰이는 모델이 분리가 되면 좋을 거 같아요. 그렇지 않으면 변경 의존도가 너무 커져요. 그리고 Dto 내부에 toEntity() 와 같은 메소드를 정의하면 비즈니스 로직에서 변환 과정이 생략될 수 있어서 가독성이 더 높아져요

Reply:
controller에서 requestbody로 MemberLoginRequestDto를 받고, 그 dto의 비밀번호를 암호화한 비밀번호로 치환한 뒤에  service 단에 넘겨 repository에 저장하려고 합니다. 그렇게 하려면 어떻게 해야 하나요? 하나의 값만 바꾸는 것이라서 setter를 쓸까 하다가도 비밀번호이기 때문에 setter는 쓰기 망설여집니다..